endpoint/src/main.rs

```diff
@@ -65,35 +76,73 @@ fn oauth2_create_token(
 }
 
 #[derive(Debug)]
-pub struct Host<'r>(Option<&'r str>);
+pub struct Host(Option<String>);
```

Comment:
Rocket has a `Host` struct with matching `FromRequest` (https://api.rocket.rs/v0.5-rc/rocket/http/uri/struct.Host.html). let's aim for replacing this custom struct with the Rocket one after merging this PR. WDYT, @fkettelhoit

Reply:
added this as #6